src/Variable.cpp

```diff
@@ -544,6 +544,32 @@ bool matioCpp::Variable::isValid() const
     return m_handler->get() && checkCompatibility(m_handler->get(), m_handler->variableType(), m_handler->valueType());
 }
 
+matioCpp::Variable matioCpp::Variable::operator[](const std::string& el)
+{
+    if (variableType() != matioCpp::VariableType::Struct)
+    {
+        std::cerr << "[ERROR][matioCpp::Variable::operator[]] The operator[](string) can be used only with structs." << std::endl;
+        assert(false);
+        return matioCpp::Variable();
+    }
+    size_t index = getStructFieldIndex(el);
+    assert(index < getStructNumberOfFields() && "The specified field does not exist.");
+    return getStructField(index);
+}
+
+const matioCpp::Variable matioCpp::Variable::operator[](const std::string& el) const
+{
+    if (variableType() != matioCpp::VariableType::Struct)
+    {
+        std::cerr << "[ERROR][matioCpp::Variable::operator[]] The operator[](string) can be used only with structs." << std::endl;
+        assert(false);
+        return matioCpp::Variable();
+    }
+    size_t index = getStructFieldIndex(el);
+    assert(index < getStructNumberOfFields() && "The specified field does not exist.");
```

Comment:
There are other `if` cases, and access speed is not really a concern here. I will add the `if` with the message

Reply:
Done in https://github.com/ami-iit/matio-cpp/pull/83/commits/554b73328e5818ea9c5cbcdb3e4733329bb07488

I had to refactor a bit some internal logic to avoid calling some matio functions multiple times.